### PR TITLE
Use a std::unique_ptr for the IR Builder

### PR DESCRIPTION
**Pull request by abadams**
instead of a raw owning pointer

### CODE REVIEW

File: src/CodeGen_LLVM.cpp [hunk 1]
```diff
@@ -287,8 +287,7 @@ void CodeGen_LLVM::initialize_llvm() {
 
 void CodeGen_LLVM::init_context() {
     // Ensure our IRBuilder is using the current context.
-    delete builder;
-    builder = new IRBuilder<>(*context);
+    builder.reset(new IRBuilder<>(*context));
```
Comment (steven-johnson) [inline src/CodeGen_LLVM.cpp:290]: Suggestion: use `std::make_unique` instead? (Your code isn't wrong but as of C++14 make_unique is in theory the 'preferred' way to do this)

File: src/CodeGen_LLVM.cpp [hunk 2]
```diff
@@ -978,7 +973,7 @@ llvm::Function *CodeGen_LLVM::add_argv_wrapper(llvm::Function *fn,
     std::vector<llvm::Value *> wrapper_args;
     for (llvm::Function::arg_iterator i = fn->arg_begin(); i != fn->arg_end(); i++) {
         // Get the address of the nth argument
-        llvm::Value *ptr = CreateConstGEP1_32(builder, i8_t->getPointerTo(),
+        llvm::Value *ptr = CreateConstGEP1_32(builder.get(), i8_t->getPointerTo(),
```
Comment (steven-johnson) [inline src/CodeGen_LLVM.cpp:976]: I'm surprised the `.get()` is necessary, doesn't a unique_ptr have an implicit conversion?

**Review by steven-johnson — APPROVED** [review]
LGTM but IMHO using `std::make_unique` would be preferable

File: src/CodeGen_LLVM.cpp (hunk 2 above)
Comment (abadams) [reply src/CodeGen_LLVM.cpp:976]: Apparently not. I get compiler errors without it.

File: src/CodeGen_LLVM.cpp (hunk 1 above)
Comment (abadams) [reply src/CodeGen_LLVM.cpp]: Done

**Pull request merged**